### PR TITLE
feature/PEER-191-1: Email notifications

### PR DESCRIPTION
**Pull request by alexeishulga**

### CODE REVIEW

File: app/containers/ChangeEmail/ConfirmEmailForm.jsx [hunk 1]
```diff
@@ -22,41 +22,92 @@ const ConfirmEmailForm = ({
...
         <Field
           name={CODE_FIELD}
           disabled={confirmOldEmailProcessing}
           label={t('signUp.verificationCode')}
           component={TextInputField}
           validate={required}
           warn={required}
+          onChange={(e) =>
+            setEncorrectCode(verificationCode !== e.target.value ? false : true)
+          }
```
Comment (AweAwe95) [inline app/containers/ChangeEmail/ConfirmEmailForm.jsx:85]: declare a separate function

File: app/components/ToogleSwitch/ToggleSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+type ToggleSwitchProps = {
+  isToggled: boolean;
+  setIsToggled: (isToggled: boolean) => void;
+};
+
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  isToggled,
+  setIsToggled,
+}): JSX.Element => {
+  const onToggle = () => setIsToggled(!isToggled);
```
Comment (almondy22) [inline app/components/ToogleSwitch/ToggleSwitch.tsx:13]: Wrap this handler in `useCallback`

File: app/containers/ChangeEmail/ConfirmEmailForm.jsx [hunk 3]
```diff
@@ -22,41 +22,92 @@ const ConfirmEmailForm = ({
   confirmOldEmail,
   confirmOldEmailProcessing,
   sendAnotherCode,
+  closeModal,
+  emailAddress,
+  verificationCodeError,
+  verificationCode,
 }) => {
+  const [encorrectCode, setEncorrectCode] = useState(verificationCodeError);
```
Comment (almondy22) [inline app/containers/ChangeEmail/ConfirmEmailForm.jsx:30]: Typo, change to `incorrect`

File: app/containers/ChangeEmail/ConfirmEmailForm.jsx [hunk 4]
```diff
@@ -22,41 +22,92 @@ const ConfirmEmailForm = ({
...
+      >
         <Field
           name={CODE_FIELD}
           disabled={confirmOldEmailProcessing}
           label={t('signUp.verificationCode')}
           component={TextInputField}
           validate={required}
           warn={required}
+          onChange={(e) =>
+            setEncorrectCode(verificationCode !== e.target.value ? false : true)
```
Comment (almondy22) [inline app/containers/ChangeEmail/ConfirmEmailForm.jsx:84]: Just `setIncorrectCode(verificationCode === e.target.value)`

File: app/containers/ChangeEmail/index.js [hunk 5]
```diff
@@ -42,62 +43,92 @@ import {
...
+            verificationCodeError={verificationCodeError}
+            verificationCode={verificationCode}
+          />
+        )}
+
+        {content === CHANGE_EMAIL_FORM && <ChangeEmailForm />}
+      </Modal>
+      {!open ? (
+        <Button
+          onClick={(e) => setEmailAddress(e.target.form[0].value)}
```
Comment (almondy22) [inline app/containers/ChangeEmail/index.js:105]: Make separate handler function and wrap it in `useCallback`

Comment (almondy22) [reply app/containers/ChangeEmail/index.js:105]: Also look for other places where you have same thing and fix it

**Review by almondy22 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**